pycaret/internal/ts_tests.py

```diff
@@ -0,0 +1,227 @@
+    from statsmodels.tsa.stattools import acf, pacf
+
+    acf_x, ci = acf(ts, alpha=0.05, nlags=20)
+    lower = ci[:, 0] - acf_x
+    upper = ci[:, 1] - acf_x
+    acf_x = np.round(((acf_x >= upper) + (acf_x <= lower)) * acf_x, 2)
+    acf_map = {val: idx for idx, val in enumerate(acf_x)}
+    acf_values = list(filter(lambda x: abs(x) > 0, acf_x))
+    acf_lags = list(map(lambda x: acf_map[x], acf_values))
+
+    pacf_x, ci = pacf(ts, alpha=0.05, nlags=20)
+    lower = ci[:, 0] - pacf_x
+    upper = ci[:, 1] - pacf_x
+    pacf_x = np.round(((pacf_x >= upper) + (pacf_x <= lower)) * pacf_x, 2)
+    pacf_map = {val: idx for idx, val in enumerate(pacf_x)}
+    pacf_values = list(filter(lambda x: abs(x) > 0, pacf_x))
+    pacf_lags = list(map(lambda x: pacf_map[x], pacf_values))
+
+    if plots:
+        f, ax = plt.subplots(nrows=2, ncols=1)
+
+        from statsmodels.graphics.tsaplots import plot_pacf, plot_acf
+        plot_acf(ts, lags=20, ax=ax[0])
+        plot_pacf(ts, lags=20, ax=ax[1])
+        f.tight_layout()
+        plt.show()
+    result = {
+        'Significant ACF lags': acf_lags,
+        'Significant PACF lags': pacf_lags,
+        'Highest ACF lag': acf_map[np.max(acf_values[1:])],
+        'Highest PACF lag': pacf_map[np.max(pacf_values[1:])],
+        '_ACF values': acf_values,
+        '_PACF values': pacf_values,
+    }
+
+    return result
+
+
+def stationarity_tests(
+    ts: Union[pd.DataFrame, pd.Series],
+    cutoff=0.05,
+    **kwargs
+):
+    from statsmodels.tsa.stattools import adfuller
+    dftest = adfuller(ts, autolag='AIC', maxlag=20)
+    pvalue = dftest[1]
+    result = {
+        'ADF stationarity': 'Difference Stationary' if pvalue < cutoff else 'Not stationary. Try differencing.',
+        '_ADF pvalue': pvalue
+    }
+
+    from statsmodels.tsa.api import kpss
+    statistic, p_value, n_lags, critical_values = kpss(ts, regression='ct', nlags='auto')
+
+    result.update({
+        'KPSS stationarity': "Not stationary. Try removing trend." if statistic > critical_values[
+            "5%"] else 'Trend Stationary',
+        '_KPSS pvalue': p_value
+    })
+
+    if result['KPSS stationarity'] == 'Trend Stationary' and result['ADF stationarity'] == 'Difference Stationary':
+        result['Stationarity'] = 'Stationary'
+    elif result['KPSS stationarity'] != 'Trend Stationary' and result['ADF stationarity'] != 'Difference Stationary':
+        result['Stationarity'] = 'Not stationary. Try GARCH models.'
+    else:
+        result['Stationarity'] = 'Partial stationarity'
+    return result
+
+
+def ts_tests(
+    ts: pd.DataFrame,
+    show_hidden_keys: bool = False,
+    **kwargs
+):
+    """
+    This method performs all statistical tests for the given time series dataset. This is further help in modelling
+    and most modelling inference
+
+    Returns
+    -------
+    Dataframe with the results for the tests
+
+    """
+    test_results = {
+        'Temporal': trend_and_seasonality_test(ts, **kwargs),
```

Comment:
If `ts_tests` is the main interface to the user, that is, `trend_and_seasonality_tests`, `correlation_tests` are not directly called by the user, then I suggest to use reference here private methods 

```python
def _trend_and_seasonality_tests():
     pass
```

and then in the main interface 

```python
 test_results = {
        'Temporal': _trend_and_seasonality_test(ts, **kwargs),
```